umh.docs.umh.app/content/en/docs/features/connectivity/additionalconnectivity/barcodereader-retrofitting.md

```diff
@@ -25,13 +25,27 @@ something like a quality testing station (we once connected a Mitutoyo quality t
 
 To use the microservice barcode reader, you will need configure the helm-chart and enable it.
 
-1. Enable \_000_commonConfig.datasources.barcodereader.enabled in the [Helm Chart](https://umh.docs.umh.app/docs/architecture/helm-chart/#configuration-options)
-2. During startup, it will show all connected USB devices. Remember yours and then change the INPUT_DEVICE_NAME and INPUT_DEVICE_PATH
-3. Also set ASSET_ID, CUSTOMER_ID, etc. as this will then send it into the topic ia/ASSET_ID/.../barcode
-4. Restart the pod
-5. Scan a device, and it will be written into the topic xxx
-
-Once installed, you can [configure](/docs/architecture/microservices/community/barcodereader/) the microservice by
+1. Enable the barcodereader feature by executing the following command:
+    ```bash
+    sudo $(which helm) upgrade --kubeconfig /etc/rancher/k3s/k3s.yaml  -n united-manufacturing-hub united-manufacturing-hub united-manufacturing-hub/united-manufacturing-hub --set _000_commonConfig.datasources.barcodereader.enabled=true --reuse-values --version $(sudo $(which helm) l
+    s --kubeconfig /etc/rancher/k3s/k3s.yaml  -n united-manufacturing-hub -o json | jq -r '.[0].app_version')
+    ```
+2. During startup, it will show all connected USB devices. Remember yours and then change the `INPUT_DEVICE_NAME` and `INPUT_DEVICE_PATH`. Also set `ASSET_ID`, `CUSTOMER_ID`, etc. as this will then send it into the topic `ia/ASSET_ID/.../barcode`. To configure these values, you should modify the barcodereader's deployment. The following command allows you for editting:
+    ```bash
+    sudo $(which kubectl) edit deployment united-manufacturing-hub-barcodereader -n united-manufacturing-hub  --kubeconfig /etc/rancher/k3s/k3s.yaml
```

Comment:
will this not be overwritten by a new helm chart version? as the helm chart will always overwrite a deployment or statefulset? @DaruZero

Reply:
I commented also [here](https://github.com/united-manufacturing-hub/umh.docs.umh.app/pull/186#discussion_r1414152577) that we cannot configure via values.yaml in the current version, so I chose the `kubectl edit`,  but it will be overwritten by `helm upgrade`. Is there any other ways to modify these configurations?

Reply:
Actually, it is possible to set those values via the `values.yaml` file, thus making it possible to use `helm upgrade` instead of `kubectl edit`. You can find the correct fields [in the documentation](https://umh.docs.umh.app/docs/reference/helm-chart/#barcode-reader)

Reply:
I updated it and tested the new command. Also, the change will be kept after `helm upgrade ... --reuse-values ...`